### ISSUE TITLE
Feign dependency of 8.17.0 causes Atlas failure in HA mode

### ISSUE THREAD

**Issue by ellisjoe** [palantir/atlasdb#648]
Our Feign dependency of 8.17.0 causes Atlas to fail during HA startup. This is due to the breaking change in Feign introduced here https://github.com/Netflix/feign/pull/390/files to using OkHttp 3.x. We'd like Atlas to upgrade Feign and OkHttp.

@uschi2000 @markelliot @briandorne

**Comment by uschi2000** [comment]
We need to fix that this week in order to unblock an Elements platform release and Papaya upgrade. Suggestions:
(1) AtlasDB upgrades to feign 8.17.0 and okhttp3
(2) AtlasDB shades feign&okhttp
(3) 1&2

**Comment by briandorne** [comment]
@ellisjoe @uschi2000 -- Are one of you the owners for submitting a PR for this?

**Comment by ellisjoe** [comment]
PR for solution 1: https://github.com/palantir/atlasdb/pull/650

**Comment by divath** [comment]
(1) Re: Won't this cause the same problems in the future, e.g. once http-remoting start using more recent okhttp?

**Comment by markelliot** [comment]
Yep, it's a temporary solution and good hygiene, real solution involves what's mentioned on the issue, which is likely shading the dependency.

As an alternative, we can rearchitect these things to be injected instead of instantiation.

**Comment by gsheasby** [comment]
Closing as won't fix as per email discussion; will open another ticket once we have an approach for solving the issue.
